packages/block-library/src/site-logo/index.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * Server-side rendering of the `core/site-logo` block.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Renders the `core/site-logo` block on the server.
+ *
+ * @return string The render.
+ */
+function render_block_core_site_logo( $attributes ) {
+	$image_attrs_filter = function ( $image_attrs ) use ( $attributes ) {
+		return $image_attrs;
+	};
+
+	$adjust_width_height_filter = function ( $image ) use ( $attributes ) {
+		if ( empty( $attributes['width'] ) ) {
+			return $image;
+		}
+		$height = floatval( $attributes['width'] ) / ( floatval( $image[1] ) / floatval( $image[2] ) );
+        return array( $image[0], intval( $attributes['width'] ), intval( $height ) );
+    };
+
+	add_filter( 'wp_get_attachment_image_src', $adjust_width_height_filter );
+
+	$custom_logo = get_custom_logo();
+	if ( ! empty( $attributes['align'] ) && in_array( $attributes['align'], array( 'center', 'left', 'right' ) ) ) {
+		$custom_logo = str_replace(
+			'class="custom-logo-link"',
+			"class=\"custom-logo-link align{$attributes['align']}\"",
+			$custom_logo
+		);
+		$html = sprintf( '<div class="wp-block-custom-logo">%s</div>', $custom_logo );
+	} else {
+		$html = str_replace(
+			'class="custom-logo-link"',
+			'class="wp-block-custom-logo custom-logo-link"',
+			$custom_logo
+		);
+	}
+	remove_filter( 'wp_get_attachment_image_src', $adjust_width_height_filter );
+
+	return $html;
```

Comment:
I didn't check  but how close is the generated  markup here compared to the editor markup?

Reply:
I just made some changes to bring them into better alignment.